# HG changeset patch
# User Marek Marczykowski <marmarek@mimuw.edu.pl>
# Date 1315172996 -7200
# Node ID 4683409fac3d4fef836ce78f01f12b31f5f9e12a
# Parent  6239209bb560b4931d4d97456c82c1a5ca4bd10a
libxl: fix double free at get_all_assigned_devices

Do not free() list manually - it will be freed by libxl__free_all.

Signed-off-by: Marek Marczykowski <marmarek@mimuw.edu.pl>

diff -r 6239209bb560 -r 4683409fac3d tools/libxl/libxl_pci.c
--- a/tools/libxl/libxl_pci.c	Wed Aug 31 15:32:47 2011 +0100
+++ b/tools/libxl/libxl_pci.c	Sun Sep 04 23:49:56 2011 +0200
@@ -469,7 +469,6 @@ static int get_all_assigned_devices(libx
     }
 
     if ( 0 == *num ) {
-        free(pcidevs);
         pcidevs = NULL;
     }else{
         *list = pcidevs;
